Conceptpower+Spring/src/main/java/edu/asu/conceptpower/rest/ConceptSearch.java

```diff
@@ -56,15 +60,19 @@
 		Map<String, String[]> queryParams = req.getParameterMap();
 		Map<String, String> searchFields = new HashMap<String, String>();
 
-		String operator = SearchParamters.OP_OR;
-		for (String key : queryParams.keySet()) {
-			if (key.trim().equals(SearchParamters.OPERATOR)
-					&& !queryParams.get(key)[0].trim().isEmpty())
-				operator = queryParams.get(key)[0].trim();
-			else {
-				searchFields.put(key.trim(), queryParams.get(key)[0]);
-			}
-		}
+        String operator = SearchParamters.OP_OR;
+        for (String key : queryParams.keySet()) {
+            if (key.trim().equals(SearchParamters.OPERATOR) && !queryParams.get(key)[0].trim().isEmpty())
+                operator = queryParams.get(key)[0].trim();
+            else {
+                // If the value is typeURI trim and add it to typeId
+                if (key.trim().equalsIgnoreCase(SearchFieldNames.TYPE_URI)) {
+                    searchFields.put("type_id", uriHelper.getTypeId(queryParams.get(key)[0]));
+                } else {
+                    searchFields.put(key.trim(), queryParams.get(key)[0]);
+                }
+            }
```

Comment:
I'd turn this into and if/elseif/else statement instead of a nested if/else.

also, I think the indentations are wrong?